tests/framework/population/test_population_view.py

```diff
@@ -121,20 +121,92 @@ def test_get_attributes_empty_idx(pies_and_cubes_pop_mgr: PopulationManager) ->
     assert pop.empty
 
 
-@pytest.mark.skip(reason="TODO [MIC-6609] add tests for empty attributes and/or query string")
-def test_get_attributes_empty_attributes_query() -> None:
-    pass
-
-
 def test_get_attributes_raises(pies_and_cubes_pop_mgr: PopulationManager) -> None:
     pv = pies_and_cubes_pop_mgr.get_view(PieComponent())
+    index = pd.Index([])
 
-    # Unknown columns
     with pytest.raises(
         PopulationError,
         match="Requested attribute\(s\) \{'foo'\} not in population table.",
     ):
-        pv.get_attributes(pd.Index([]), "foo")
+        pv.get_attributes(index, "foo")
+
+    with pytest.raises(
+        PopulationError,
+        match="provide the columns needed to evaluate that query",
+    ):
+        pv.get_attributes(index, [], "pie == 'apple'")
+
+
+@pytest.mark.parametrize(
+    "pv_query",
+    [
+        None,
+        "pi > 10",
+        "pie == 'chocolate'",
+        "cube > 20000",
+        "pi < 10000 and pie != 'apple' and cube < 40000",
+    ],
+)
+@pytest.mark.parametrize(
+    "additional_query",
+    [
+        None,
+        "pi < 3000",
+        "pie != 'pumpkin'",
+        # We can also filter by public columns
+        "10000 <= cube <= 40000",
+        "pi > 3000 and (pie == 'apple' or pie == 'sweet_potato') and cube > 15000",
+    ],
+)
+def test_get_attributes_combined_query(
+    pv_query: str | None,
+    additional_query: str | None,
+    pies_and_cubes_pop_mgr: PopulationManager,
+) -> None:
+    """Test that queries provided to the pop view and via get_attributes are combined correctly."""
+    pv_kwargs = {}
+    if pv_query is not None:
+        pv_kwargs["query"] = pv_query
+    pv = pies_and_cubes_pop_mgr.get_view(PieComponent(), **pv_kwargs)
+    full_idx = pd.RangeIndex(0, len(PIE_RECORDS))
+    kwargs = {}
+    if additional_query is not None:
+        kwargs["query"] = additional_query
+
+    combined_query_parts = []
+    if pv_query is not None:
+        combined_query_parts.append(f"{pv_query}")
+    if additional_query is not None:
+        combined_query_parts.append(f"{additional_query}")
+    combined_query = " and ".join(combined_query_parts)
+
+    col_request = PIE_COL_NAMES.copy()
+    if combined_query and "cube" in combined_query:
+        col_request += ["cube"]
+    pop = pv.get_attributes(full_idx, col_request, **kwargs)  # type: ignore[arg-type]
+
+    expected_pop = PIE_DF.copy()
```

Comment:
I don't think we need this copy or the one on line 192, since we never modify `expected_pop` in-place.